src/update/verify_test.go

```diff
@@ -2,67 +2,78 @@ package update
 
 import (
 	"bytes"
+	"crypto"
 	"io"
 	"os"
+	"path/filepath"
 	"testing"
 
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
+	"github.com/sigstore/sigstore/pkg/signature"
 	"github.com/stretchr/testify/assert"
 )
 
+var goodKey = loadKey("good.pem")
+var badKey = loadKey("bad.pem")
+
+var goodKeyPub = loadKey("good.pub.pem")
+
+func loadKey(name string) []byte {
+	key, err := os.ReadFile(filepath.Join("src/update/test_data", name))
+	if err != nil {
+		panic(err)
+	}
+	return key
+}
+
+func sign(key []byte, msg []byte) io.Reader {
+	priv, err := cryptoutils.UnmarshalPEMToPrivateKey(key, cryptoutils.SkipPassword)
+	if err != nil {
+		panic(err)
+	}
+
+	verifier, err := signature.LoadSignerVerifier(priv, crypto.SHA256)
+	if err != nil {
+		panic(err)
+	}
+
+	bs, err := verifier.SignMessage(bytes.NewReader(msg))
+	if err != nil {
+		panic(err)
+	}
+
+	return bytes.NewReader(bs)
+}
+
 // readFile reads a file into an io.Reader
 // It uses os.ReadFile because that more closely mimics how we would do this
 // for real (i.e. we would read to a buffer over HTTP then verify that, because we
 // need to reuse the reader again afterwards and don't want to parse the tarball
 // until we're sure it's OK).
-func readFile(filename string) io.Reader {
+func readFile(filename string) []byte {
 	b, err := os.ReadFile(filename)
 	if err != nil {
 		log.Fatalf("%s", err)
 	}
-	return bytes.NewReader(b)
-}
-
-func TestVerifyGoodSignature(t *testing.T) {
-	signed := readFile("src/update/test_data/test.txt")
-	signature := readFile("src/update/test_data/test.txt.asc")
-	assert.True(t, verifySignature(signed, signature))
-}
-
-func TestVerifyBadSignature(t *testing.T) {
-	signed := readFile("src/update/test_data/test.txt")
-	signature := readFile("src/update/test_data/bad.txt.asc")
-	assert.False(t, verifySignature(signed, signature))
+	return b
 }
 
-func TestVerifyBadFile(t *testing.T) {
-	signed := readFile("src/update/test_data/bad.txt")
-	signature := readFile("src/update/test_data/test.txt.asc")
-	assert.False(t, verifySignature(signed, signature))
+func fileReader(filename string) io.Reader {
+	return bytes.NewReader(readFile(filename))
```

Comment:
Seems a bit circular to slurp an `io.Reader` into a byte slice and then return an `io.Reader` backed by that byte slice - could you just use `os.Open` instead? (It probably doesn't matter though, it doesn't look like we're reading large files anyway.)

Reply:
There's a circleci joke in there somewhere